lib/krb5/pac.c

```diff
@@ -911,6 +932,21 @@ krb5_pac_verify(krb5_context context,
 			      privsvr);
 	if (ret)
 	    return ret;
+
+	if (pac->ticket_sign_data.length != 0) {
+
+	    if (pac->ticket_checksum == NULL) {
+		krb5_set_error_message(context, EINVAL,
+				       "PAC missing ticket-checksum");
+		return EINVAL;
+	    }
+
+	    ret = verify_checksum(context, pac->ticket_checksum, &pac->data,
```

Comment:
BTW do we want to defense in depth and also check for keyed checksums in `create_checksum()`?

Reply:
The defence in depth I really want is to remove unkeyed checksums as 'weak', but until then I would support such a change.

Reply:
Ins't it checked elsewhere? I'll look.
BTW, can't we get rid of unkeyed checksums altogether already? Wasn't it only used for DES enctypes?

Reply:
@abartlet I didn't see your comment while writing mine, but I'm not surprise we think alike on this :)

Reply:
Unkeyed SHA checksums are used by PKINIT (`build_auth_pack()`) and GSS pre-auth `pa_gss_decode_context_state()` so they can't be gotten rid of entirely. Let me discuss this off-list.

Reply:
Leaving this out given your related, more generic work.